server/src/storage/datastore.ts

```diff
@@ -33,4 +34,27 @@ const getWithId = async (kind: string, id: string) => {
   };
 };
 
-export {getWithId};
+/**
+ * A Datastore wrapper that inserts a particular entity with the specified Kind.
+ * @param kind The Kind of the Entity
+ * @param data The data of the Entity to be added
+ */
+const add = async (kind: string, data: object) => {
+  const key = datastore.key(kind);
+  const entity = {key, data};
+  const [res] = await datastore.insert(entity);
+  return getIdFromCommitResponse(res);
+};
+
+/**
+ * Unpacks id of inserted object from CommitResponse object received from Datastore.
+ */
+const getIdFromCommitResponse = (res: google.datastore.v1.ICommitResponse) => {
+  const mutationResult = res.mutationResults && res.mutationResults[0];
```

Comment:
Can you use optional chaining? https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#optional-chaining

Reply:
Okay!